bin/varnishadm/varnishadm.c

```diff
@@ -433,7 +437,13 @@ pass(int sock)
 	VCLS_AddFunc(vcls, 1, cli_askmgt);
 	VCLS_SetHooks(vcls, NULL, vadm_cli_cb_after);
 	do {
-		i = VCLS_Poll(vcls, cli, -1);
+		i = VCLS_Poll(vcls, cli, 0);
+		j = poll(fds, 1, 0);
+		if (j == 1 && recv(sock, buf, 1, MSG_PEEK) == 0) {
+			fprintf(stderr, "CLI communication error\n");
+			RL_EXIT(1);
+		}
```

Comment:
I used peek, because, on my system at least, poll would report POLLIN event when the remote socket is closed, and that could not be differentiated from when we have actual data to read (which should never be the case but still..), so I found using recv with MSG_PEEK to be the most reliable solution here as it will always return 0 on EOF